VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -46,60 +38,52 @@ public void render(BaseViewHolder tag, DiscussionThread discussionThread) {
         String threadTitle = discussionThread.getTitle();
         holder.discussionPostTitle.setText(threadTitle);
 
-        if (discussionThread.isPinned() || discussionThread.isFollowing()) {
-            holder.discussionPostPinFollowIcon.setVisibility(View.VISIBLE);
-            holder.discussionPostPinFollowTextView.setVisibility(View.VISIBLE);
-
-            Iconify.IconValue pinFollowIcon = discussionThread.isPinned() ?
-                    Iconify.IconValue.fa_pinterest : Iconify.IconValue.fa_star;
-            holder.discussionPostPinFollowIcon.setIcon(pinFollowIcon);
-
+        if (discussionThread.getAuthorLabel() != null) {
+            holder.discussionPostPinTextView.setVisibility(View.VISIBLE);
             String pinFollowTextLabel = "";
-            if (discussionThread.isFollowing()) {
-                pinFollowTextLabel = context.getString(R.string.discussion_posts_following);
-
-            } else if (discussionThread.getAuthorLabel() == PinnedAuthor.STAFF) {
-                pinFollowTextLabel = context.getString(R.string.discussion_priviledged_author_label_staff);
+            if (discussionThread.getAuthorLabel() == PinnedAuthor.STAFF) {
```

Comment:
This could be converted into a `switch` statement, which would avoid calling the `getAuthorLabel()` twice.